File: Problem36/Double_base_pal.java
import java.math.BigInteger;

public class Double_base_pal {

    public static void main(String args[]) {

        int limit = 1000000, sum = 0;

        for(int j = 1; j<limit; j+=2){
            BigInteger k = BigInteger.valueOf(j);
	    BigInteger binar = binary(j);
	    if(CheckPalindrome(k) == true)
		if(CheckPalindrome( binar ) == true)
		    sum+=j;
	}
	System.out.println("sum = " + sum);
    }

    public static boolean CheckPalindrome(BigInteger i){
        BigInteger j = BigInteger.valueOf(0), remainder;
        BigInteger hold = i;
        // compare bi1 with bi2
        while( i.equals(BigInteger.valueOf(0) ) == false ){
            // perform remainder operation on i using 10
            remainder = i.remainder(BigInteger.valueOf(10));
            // multiply j with 10 and assign result to j
            j = j.multiply(BigInteger.valueOf(10));
            // perform add operation on j using remainder
            j = j.add(remainder);
            // divide i with 10
            i = i.divide(BigInteger.valueOf(10));
        }
        if (j.equals(hold) == true) return true;
        else return false;
    }

    public static BigInteger binary(int i){
        BigInteger bina = BigInteger.valueOf(0);
        BigInteger bina2 = BigInteger.valueOf(0);
        BigInteger ten = BigInteger.valueOf(10);
        BigInteger two = BigInteger.valueOf(2);

        int n;

        double M = 0.30102999566;
        while(i > 0){
	    // 4, 2
            n = (int) (Math.log10(i)/M);
	    //System.out.println("n = " + n);
            bina = ten.pow(n);  // 10000, 10100
	    //System.out.println("bina = " + bina);
            bina2 = bina2.add(bina);
	    //System.out.println("bina2 = " + bina2);
            i = (int)(i - Math.pow(2, n)); // 7, 2
        }
        //System.out.println("binary = " + bina2);
        return bina2;
    }
}

File: Problem36/Double_base_pal.c
#include<stdio.h>
#include<math.h>
#define true 1
#define false 0
#define M 0.30102999566

long long int binary(long long int i){
  // i = 23
  long long int n, bina = 0;

  while(i > 0){
    n = log10(i)/M; // 4, 2
    bina += powl(10, (double)n); // 10000, 10100
    i = i - powl(2, (double)n); // 7, 2
  }
  return bina;
}

long long int CheckPalindrome(long long int i){

  long long int j = 0, hold, remainder;
  hold = i;

  while(i != 0){
    remainder = i%10;
    j = j*10 + remainder;
    i = i/10;
  }

  if(j == hold) return true;
  else return false;

}

int main(){

  long long int i = 584;
  long long int j;
  long long int sum = 0;
  long long int limit = 1000000;

  printf("binary = %lld\n", binary(limit - 1));
  getchar();

  for(j = 1; j<limit; j+=2){
    if(CheckPalindrome(j) == true)
      if(CheckPalindrome(binary(j)) == true){
	sum+=j;
	printf("j = %lld\n", j);
	printf("binary = %lld\n", binary(j));
	printf("sum = %lld\n", sum);
	getchar();
      }
  }
  printf("sum = %lld\n", sum);
}

//9223372036854775807
//10011100010000
//10011100010000
//11000011010011111
//10010000000001001
